app/assets/javascripts/discourse/app/components/form-template-field/upload.gjs

```diff
@@ -0,0 +1,122 @@
+import Component from "@ember/component";
+import UppyUploadMixin from "discourse/mixins/uppy-upload";
+import { computed } from "@ember/object";
+import { htmlSafe } from "@ember/template";
+import { tracked } from "@glimmer/tracking";
+import { dasherize } from "@ember/string";
+import { isAudio, isImage, isVideo } from "discourse/lib/uploads";
+import PickFilesButton from "discourse/components/pick-files-button";
+import icon from "discourse-common/helpers/d-icon";
+
+export default class FormTemplateFieldUpload extends Component.extend(
+  UppyUploadMixin
+) {
+  <template>
+    <div class="control-group form-template-field" data-field-type="upload">
+      {{#if @attributes.label}}
+        <label class="form-template-field__label">
+          {{@attributes.label}}
+          {{#if @validations.required}}
+            {{icon "asterisk" class="form-template-field__required-indicator"}}
+          {{/if}}
+        </label>
+      {{/if}}
+
+      {{#if @attributes.description}}
+        <span class="form-template-field__description">
+          {{htmlSafe @attributes.description}}
+        </span>
+      {{/if}}
+
+      <input type="hidden" name={{@id}} value={{this.uploadValue}} />
+
+      <PickFilesButton
+        @fileInputClass="form-template-field__upload"
+        @fileInputId={{this.fileUploadElementId}}
+        @allowMultiple={{@attributes.allow_multiple}}
+        @showButton={{true}}
+        @onFilesPicked={{true}}
+        @icon="upload"
+        @label={{this.uploadStatusLabel}}
+        @fileInputDisabled={{this.disabled}}
+        @acceptedFormatsOverride={{@attributes.file_types}}
+        @acceptedFileTypesString={{@attributes.file_types}}
+      />
+
+      {{#if this.uploadedFiles}}
+        <ul class="form-template-field__uploaded-files">
+          {{#each this.uploadedFiles as |file|}}
+            <li>
+              {{icon "file"}}
+              <a
+                href={{file.url}}
+                target="_blank"
+                rel="noopener noreferrer"
+              >{{file.file_name}}</a>
+              <span>{{file.human_filesize}}</span>
+            </li>
+          {{/each}}
+        </ul>
+      {{/if}}
+    </div>
+  </template>
+
+  @tracked uploadValue;
+  @tracked uploadComplete = false;
+  @tracked uploadedFiles = [];
+  @tracked disabled = this.uploading;
+  @tracked fileUploadElementId = `${dasherize(this.id)}-uploader`;
+  @tracked fileInputSelector = `#${this.fileUploadElementId}`;
+
+  type = "composer";
+
+  @computed("uploading", "uploadValue")
```

Comment:
Does this need to be `computed`? I think it can just be a normal tracked getter right?

Reply:
I left it unchanged, and now that I tried without it, it doesn't update the label. I believe it's because `this.uploading` comes from upstream in a non-tracked context.